src/Nest/CommonAbstractions/ConnectionSettings/ClrTypeDefaults.cs

```diff
@@ -26,6 +26,11 @@ public interface IClrTypeMapping
 		/// </summary>
 		string RelationName { get; set; }
 
+		/// <summary>
+		/// The property for <see cref="ClrType"/> to resolve ids from.
+		/// </summary>
+		string IdPropertyVerbatim { get; set; }
```

Comment:
It;s mainly because we already have `IdProperty` being `Expression<Func<TDocument, object>>` so we need two properties. Suggestions on a different name?

Reply:
OK I see, where `IClrTypeMapping<TDocument>` implements `IClrTypeMapping`. 

Maybe `IdPropertyName`, in line with `IndexName`, `TypeName` and `RelationName`?

Reply:
Better! Will amend.